drivers/clock_control/clock_control_nrf2_hfxo.c

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_nrf54h_hfxo
+
+#include "clock_control_nrf2_common.h"
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/clock_control/nrf_clock_control.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(clock_control_nrf2, CONFIG_CLOCK_CONTROL_LOG_LEVEL);
+
+#include <hal/nrf_lrcconf.h>
+
+BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 1,
+	     "multiple instances not supported");
+
+struct dev_data_hfxo {
+	struct onoff_manager mgr;
+	onoff_notify_fn notify;
+	struct k_timer timer;
+};
+
+struct dev_config_hfxo {
+	uint32_t fixed_frequency;
+	uint16_t fixed_accuracy;
+	k_timeout_t start_up_time;
+};
+
+static void hfxo_start_up_timer_handler(struct k_timer *timer)
+{
+	struct dev_data_hfxo *dev_data =
+		CONTAINER_OF(timer, struct dev_data_hfxo, timer);
+
+	/* In specific cases, the HFXOSTARTED event might not be set even
+	 * though the HFXO has started (this is a hardware issue that will
+	 * be fixed). For now, the HFXO is simply assumed to be started
+	 * after its configured start-up time expires.
+	 */
+	LOG_DBG("HFXOSTARTED: %u",
+		nrf_lrcconf_event_check(NRF_LRCCONF010,
+					NRF_LRCCONF_EVENT_HFXOSTARTED));
+
+	if (dev_data->notify) {
+		dev_data->notify(&dev_data->mgr, 0);
+	}
+}
+
+static void onoff_start_hfxo(struct onoff_manager *mgr, onoff_notify_fn notify)
+{
+	struct dev_data_hfxo *dev_data =
+		CONTAINER_OF(mgr, struct dev_data_hfxo, mgr);
+	const struct device *dev = DEVICE_DT_INST_GET(0);
+	const struct dev_config_hfxo *dev_config = dev->config;
+
+	dev_data->notify = notify;
+
+	nrf_lrcconf_event_clear(NRF_LRCCONF010, NRF_LRCCONF_EVENT_HFXOSTARTED);
```

Comment:
Like described in FLL16M, this must be preceded by setting `NRF_LRCCONF010->POWERON.MAIN = 1`.

Reply:
Added in new PR